src/spatial_reaction_systems/spatial_reactions.jl

```diff
@@ -0,0 +1,114 @@
+### Spatial Reaction Structures ###
+
+# Abstract spatial reaction structures.
+abstract type AbstractSpatialReaction end
+
+### EdgeParameter Metadata ###
+
+# Implements the edgeparameter metadata field.
+struct EdgeParameter end
+Symbolics.option_to_metadata_type(::Val{:edgeparameter}) = EdgeParameter
+
+# Implements the isedgeparameter check function.
+isedgeparameter(x::Num, args...) = isedgeparameter(Symbolics.unwrap(x), args...)
+function isedgeparameter(x, default = false)
+    p = Symbolics.getparent(x, nothing)
+    p === nothing || (x = p)
+    Symbolics.getmetadata(x, EdgeParameter, default)
+end
+
+### Transport Reaction Structures ###
+
+# A transport reaction. These are simple to handle, and should cover most types of spatial reactions.
+# Only permit constant rates (possibly consisting of several parameters).
+struct TransportReaction <: AbstractSpatialReaction
+    """The rate function (excluding mass action terms). Currently only constants supported"""
+    rate::Any
+    """The species that is subject to diffusion."""
+    species::BasicSymbolic{Real}
+
+    # Creates a diffusion reaction.
+    function TransportReaction(rate, species)
+        new(rate, species.val)
```

Comment:
I thought I've mentioned this several times? Because spatial reactions are not created in the context of a reaction system, it is not possible to know what is a parameter or a species.

Having spatial reactions tied to a (non-spatial) `ReactionSystem` could make sense, and I think a future feature would be to allow creating these a part of the `ReactionSystem`, e.g.
```
@reaction_network begin
    @transport_reactions begin
         D, X
     end
     (p,d), 0 <--> X
```

However, if we want to permit spatial reactions to be created independently of `ReactionSystem`s (which I currently think we should), and potentially be applied to different `ReactionSystem`s, then the check will have to happen when they are combined to a `LatticeReactionSystem`.